src/hashcomp.cpp

```diff
@@ -273,15 +283,15 @@ bool irc::sepstream::GetToken(std::string &token)
 		this->pos = this->tokens.find_first_not_of(this->sep, this->pos);
 		if (this->pos == std::string::npos)
 		{
-			this->pos = this->tokens.length() + 1;
+			this->pos = this->tokenlen + 1;
 			token.clear();
 			return false;
 		}
 	}
 
 	size_t p = this->tokens.find(this->sep, this->pos);
 	if (p == std::string::npos)
-		p = this->tokens.length();
```

Comment:
std::string::length() is usually (the standard recommends it and its done by all of the compilers we support) O(1) so caching a string length is unnecessary.